### ISSUE TITLE
Hibernate - org.hibernate.MappingException: Unknown entity in prod mode

### ISSUE THREAD

**Issue by tistvan** [playframework/playframework#4635]
When I run my application in production mode I have the following exception: 
```
Caused by: org.hibernate.MappingException: Unknown entity: models.TestModel
	at org.hibernate.internal.SessionFactoryImpl.getEntityPersister(SessionFactoryImpl.java:1096) ~[org.hibernate.hibernate-core-4.3.9.Final.jar:4.3.9.Final]
	at org.hibernate.internal.SessionImpl.getEntityPersister(SessionImpl.java:1479) ~[org.hibernate.hibernate-core-4.3.9.Final.jar:4.3.9.Final]
	at org.hibernate.event.internal.DefaultMergeEventListener.onMerge(DefaultMergeEventListener.java:163) ~[org.hibernate.hibernate-core-4.3.9.Final.jar:4.3.9.Final]
	at org.hibernate.event.internal.DefaultMergeEventListener.onMerge(DefaultMergeEventListener.java:85) ~[org.hibernate.hibernate-core-4.3.9.Final.jar:4.3.9.Final]
	at org.hibernate.internal.SessionImpl.fireMerge(SessionImpl.java:876) ~[org.hibernate.hibernate-core-4.3.9.Final.jar:4.3.9.Final]
	at org.hibernate.internal.SessionImpl.merge(SessionImpl.java:858) ~[org.hibernate.hibernate-core-4.3.9.Final.jar:4.3.9.Final]
	at org.hibernate.internal.SessionImpl.merge(SessionImpl.java:863) ~[org.hibernate.hibernate-core-4.3.9.Final.jar:4.3.9.Final]
	at org.hibernate.jpa.spi.AbstractEntityManagerImpl.merge(AbstractEntityManagerImpl.java:1196) ~[org.hibernate.hibernate-entitymanager-4.3.9.Final.jar:4.3.9.Final]
	... 44 common frames omitted
```

Steps to reproduce:

 * Create a new play-java application
 * Add Hibernate to the project (https://www.playframework.com/documentation/2.4.x/JavaJPA)
 * Create an entity
 * Create an action which does anything with that entity (eg. select)
 * Run the application in prod mode with `activator testProd`
 * Call that action

A workaround is to add this line to persistence.xml: 
`<jar-file>target/universal/stage/lib/<organization>.<projectname>-<version>.jar</jar-file>`

**Comment by mkurz** [comment]
Duplicate of #4590 and #4594
Checkout these two issues for a workaround. The problem will be fixed in 2.4.1.

**Comment by tistvan** [comment]
thanks
